src/gui/MainWindow.cpp

```diff
@@ -530,6 +530,12 @@ void MainWindow::updateWindowTitle()
         windowTitle = QString("%1 - %2").arg(customWindowTitlePart, BaseWindowTitle);
     }
 
+    if (m_ui->tabWidget->isModified(tabWidgetIndex)) {
+        setWindowModified(true);
```

Comment:
Don't use if (condition) set something true else set something false. Use the condition directly:

```
setWindowModified(m_ui->tabWidget->isModified(tabWidgetIndex));
```